llvm/lib/SYCLLowerIR/LowerInvokeSimd.cpp

```diff
@@ -248,9 +485,15 @@ PreservedAnalyses SYCLLowerInvokeSimdPass::run(Module &M,
     for (User *Usr : Users) {
       // a call can be the only use of the invoke_simd built-in
       CallInst *CI = cast<CallInst>(Usr);
-      Modified |= processInvokeSimdCall(CI);
+      Modified |= processInvokeSimdCall(CI, ClonedHelpers);
+    }
+  }
+  for (Function *F : ClonedHelpers) {
+    if (F->getNumUses() == 0) {
+      F->eraseFromParent();
     }
   }
+  verifyModule(M, &llvm::errs());
```

Comment:
This is a debug call, which needs to be deleted, right?
```suggestion
```

Reply:
Right, thanks.